python/TestHarness/schedulers/Scheduler.py

```diff
@@ -0,0 +1,539 @@
+
+                # Skipped/Silent/Deleted Testers fall into this catagory, caused by 'job_container' being skipped
+                # during the first iteration above
+                except dag.DAGEdgeDepError:
+                    pass
+
+        # With a working DAG created above (even a partial one), discover race conditions with remaining runnable
+        # testers.
+        while temp_dag.size():
+            output_files_in_dir = set()
+
+            # Get a list of concurrent tester containers
+            concurrent_jobs = temp_dag.ind_nodes()
+
+            for job_container in concurrent_jobs:
+                tester = job_container.getTester()
+                output_files = tester.getOutputFiles()
+
+                # check if we have colliding output files
+                if len(output_files_in_dir.intersection(set(output_files))):
+
+                    # Fail this concurrent group of testers
+                    for this_job in concurrent_jobs:
+                        tester = this_job.getTester()
+                        tester.setStatus('OUTFILE RACE CONDITION', tester.bucket_fail)
+                        failed_or_skipped_testers.add(tester)
+
+                    # collisions detected, move on to the next set
+                    break
+
+                output_files_in_dir.update(output_files)
+
+            # Delete this group of job containers and allow the loop to continue
+            for job_container in concurrent_jobs:
+                temp_dag.delete_node(job_container)
+
+        return failed_or_skipped_testers
+
+    def schedule(self, testers):
+        """
+        Schedule supplied list of testers for execution.
+        """
+        # If any threads caused an exception, we have closed down the queue and need to
+        # not schedule any more jobs
+        if self.tester_pool._state:
+            return
+
+        job_dag = dag.DAG()
+        non_runnable_jobs = set([])
+        name_to_job_container = {}
+
+        # Increment our simple queue count with the number of testers the scheduler received
+        with self.slot_lock:
+            self.job_queue_count += len(testers)
+
+        # Create a local dictionary of tester names to job containers. Add this dictionary to a
+        # set. We will use this set as a way to gain access to their methods.
+        for tester in testers:
+            name_to_job_container[tester.getTestName()] = TesterData(tester, job_dag, self.options)
+            self.tester_datas.add(name_to_job_container[tester.getTestName()])
+
+        skipped_or_failed_testers = self.checkGroupFailures(name_to_job_container)
+
+        # Create a set of failing job containers
+        for failed_tester in skipped_or_failed_testers:
+            non_runnable_jobs.add(name_to_job_container[failed_tester.getTestName()])
+
+        # Create dependencies in the DAG shared between this group of testers
+        self.createDependencies(name_to_job_container)
```

Comment:
Should be able to delete this line. Might need minor update so that you actually save the dag the first time you build it.